crates/bevy_render/src/shader/shader.rs

```diff
@@ -27,24 +27,59 @@ pub enum ShaderError {
     #[error("Shader compilation error: {0}")]
     Compilation(String),
 
-    #[cfg(any(target_os = "ios", all(target_arch = "aarch64", target_os = "macos")))]
+    #[cfg(not(any(
+        target_arch = "wasm32",
+        all(target_arch = "x86_64", target_os = "linux", target_env = "gnu"),
+        all(target_arch = "x86_64", target_os = "macos"),
+        all(target_arch = "aarch64", target_os = "android"),
+        all(target_arch = "armv7", target_os = "android"),
+        all(target_arch = "x86_64", target_os = "windows", target_env = "msvc"),
+        all(target_arch = "x86", target_os = "windows", target_env = "msvc"),
+        all(target_arch = "x86_64", target_os = "windows", target_env = "gnu"),
+        all(target_arch = "x86", target_os = "windows", target_env = "gnu")
```

Comment:
The Version of `glsl-to-spirv` on crates.io, does not yet have the Support for these three Targets.
Either they should be removed for now, or Cart should release a new version of `glsl-to-spirv`.

Reply:
I just tried publishing the new version, but the new crate structure causes problems: https://github.com/cart/glsl-to-spirv/issues/12.

Short term if we want to merge this asap we should assume we need to use the current version here.

Reply:
I removed them

Reply:
>  if we want to merge this
I opened this after seeing several people with issue around that, but none in the last few days.